pq-crypto/sike_r3/api.h

```diff
@@ -0,0 +1,77 @@
+/********************************************************************************************
+* Supersingular Isogeny Key Encapsulation Library
+*
+* Abstract: API header file for P434
+*********************************************************************************************/  
+
+#pragma once
+
+#include "sikep434r3.h"
+
+/*********************** Key encapsulation mechanism API ***********************/
+
+#define CRYPTO_SECRETKEYBYTES     374    // MSG_BYTES + SECRETKEY_B_BYTES + CRYPTO_PUBLICKEYBYTES bytes
+#define CRYPTO_PUBLICKEYBYTES     330
+#define CRYPTO_BYTES               16
+#define CRYPTO_CIPHERTEXTBYTES    346    // CRYPTO_PUBLICKEYBYTES + MSG_BYTES bytes
+
+// Encoding of keys for KEM-based isogeny system "SIKEp434" (wire format):
+// ----------------------------------------------------------------------
+// Elements over GF(p434) are encoded in 55 octets in little endian format (i.e., the least significant octet is located in the lowest memory address). 
+// Elements (a+b*i) over GF(p434^2), where a and b are defined over GF(p434), are encoded as {a, b}, with a in the lowest memory portion.
+//
+// Private keys sk consist of the concatenation of a 16-byte random value, a value in the range [0, 2^217-1] and the public key pk. In the SIKE API, 
+// private keys are encoded in 374 octets in little endian format. 
+// Public keys pk consist of 3 elements in GF(p434^2). In the SIKE API, pk is encoded in 330 octets. 
+// Ciphertexts ct consist of the concatenation of a public key value and a 16-byte value. In the SIKE API, ct is encoded in 330 + 16 = 346 octets.  
+// Shared keys ss consist of a value of 16 octets.
+
+/*********************** Ephemeral key exchange API ***********************/
+
+// SECURITY NOTE: SIDH supports ephemeral Diffie-Hellman key exchange. It is NOT secure to use it with static keys.
+// See "On the Security of Supersingular Isogeny Cryptosystems", S.D. Galbraith, C. Petit, B. Shani and Y.B. Ti, in ASIACRYPT 2016, 2016.
+// Extended version available at: http://eprint.iacr.org/2016/859
+
+// Generation of Bob's secret key 
+// Outputs random value in [0, 2^Floor(Log(2,3^137)) - 1] to be used as Bob's private key
+#define random_mod_order_B S2N_SIKE_P434_R3_NAMESPACE(random_mod_order_B)
+int random_mod_order_B(unsigned char* random_digits);
```

Comment:
I mentioned this in https://github.com/aws/s2n-tls/pull/2694#discussion_r616214297, but these functions really should be prefixed to ensure they stay unique for this section of the codebase. How do we ensure this stays distinct from this other function by the same name? https://github.com/aws/s2n-tls/blob/42bf4b1cb904f264913a70e8728dc7acdee26067/pq-crypto/sike_r1/sidh_r1.c#L60

Reply:
Similar to the kyber PR, I was using a macro to prefix the functions. I updated the PR description and added a comment in the code to try to make this clearer.